R/class-yarn.R

```diff
@@ -203,6 +203,55 @@ yarn <- R6::R6Class("yarn",
       self$body <- add_md(self$body, md, where)
       invisible(self)
     },
+    #' @description append abritrary markdown to a node or set of nodes
+    #'
+    #' @param md a string of markdown formatted text.
+    #' @param nodes an XPath expression that evaulates to object of class
+    #'   `xml_node` or `xml_nodeset` that are all either inline or block nodes
+    #'   (never both). The XPath expression is passed to [xml2::xml_find_all()].
+    #'   If you want to append a specific node, you can pass that node to this
+    #'   parameter.
+    #' @param space if `TRUE`, inline nodes will have a space inserted before
+    #'   they are appended.
+    #' @details this is similar to the `add_md()` method except that it can do
+    #'   the following:
+    #'   1. append content after a _specific_ node or set of nodes
+    #'   2. append content to multiple places in the document
+    #' @examples
+    #' path <- system.file("extdata", "example2.Rmd", package = "tinkr")
+    #' ex <- tinkr::yarn$new(path)
+    #' # append a note after the first heading
+    #'
+    #' txt <- c("> Hello from *tinkr*!", ">", ">  :heart: R")
+    #' ex$append_md(txt, ".//md:heading[1]")$head(20)
+    append_md = function(md, nodes = NULL, space = TRUE) {
+      self$body <- insert_md(self$body, md, nodes, where = "after", space = space)
+      invisible(self)
+    },
+    #' @description prepend abritrary markdown to a node or set of nodes
```

Comment:
```suggestion
    #' @description prepend abritrarily markdown to a node or set of nodes
```

why not document both prepend and append on the same page?